File: Leetcode/PY/Leetcode1838-最高频元素的频数.py
from typing import List
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        l,res,n = 0,1,len(nums)
        need = 0
        for r in range(1,n):
            need += (r - l) * (nums[r] - nums[r - 1])
            while need > k:
                need -= nums[r] - nums[l]
                l += 1
            res = max(res,r - l + 1)
        return res

File: Leetcode/Rust/Leetcode1838-最高频元素的频数.rs
impl Solution {
    pub fn max_frequency(nums:Vec<i32>, k: i32) -> i32 {
        let mut nums = nums;
        nums.sort_unstable();
        let(mut res,n,mut l) = (1,nums.len(),0);
        let mut need = 0;
        for r in 1..n{
            need += (r - l) as i32 * (nums[r] - nums[r - 1]);

            while need > k{
                need -= nums[r] - nums[l];
                l += 1;
            }
            res = res.max((r - l + 1) as i32);
        }
        res
    }
}